virtual_machine.go

```diff
@@ -455,21 +455,23 @@ func (v *VirtualMachine) AgentGetNetworkIFaces() (iFaces []*AgentNetworkIface, e
 }
 
 func (v *VirtualMachine) WaitForAgent(seconds int) error {
-	wait := time.Duration(seconds) * time.Second
+	timeout := time.After(time.Duration(seconds) * time.Second)
+	ticker := time.NewTicker(DefaultWaitInterval)
+	defer ticker.Stop()
+
 	for {
+		_, err := v.AgentOsInfo()
+		if err == nil {
+			return nil
+		}
+		if !strings.Contains(err.Error(), "500 QEMU guest agent is not running") {
+			return err
+		}
+
 		select {
-		case <-time.After(wait):
+		case <-timeout:
 			return ErrTimeout
-		default:
-			_, err := v.AgentOsInfo()
-			if err != nil {
-				if strings.Contains(err.Error(), "500 QEMU guest agent is not running") {
-					time.Sleep(DefaultWaitInterval)
-					continue
-				}
-				return err
-			}
-			return nil
+		case <-ticker.C:
```

Comment:
changing from `time.Sleep` mostly just to match the code in `WaitForAgentExecExit` below